.circleci/config.yml

```diff
@@ -0,0 +1,155 @@
+          command: go mod tidy
+      - run: |
+          if ! git diff --exit-code; then
+            echo "Git directory has vendor changes"
+            exit 1
+          fi
+      - notify_main_failure
+
+  go-test:
+    docker:
+      - image: *GOLANG_IMAGE
+      - image: docker.mirror.hashicorp.services/localstack/localstack:latest
+      - image: docker.mirror.hashicorp.services/letsencrypt/pebble
+        command: ["pebble"]
+        environment:
+          PEBBLE_VA_NOSLEEP: 1
+          PEBBLE_VA_ALWAYS_VALID: 1
+      - image: docker.mirror.hashicorp.services/circleci/postgres:11-alpine
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: waypoint_test
+      - image: docker.mirror.hashicorp.services/vault
+        command: ["server", "-dev", "-dev-root-token-id=hznroot"]
+    parallelism: 4
+    environment:
+      <<: *ENVIRONMENT
+      GOTAGS: ""
+      # GOMAXPROCS defaults to number of cores on underlying hardware, set
+      # explicitly to avoid OOM issues https://support.circleci.com/hc/en-us/articles/360034684273-common-GoLang-memory-issues
+      GOMAXPROCS: 4
+      # The medium resource class (default) boxes are 2 vCPUs, 4GB RAM
+      # https://circleci.com/docs/2.0/configuration-reference/#docker-executor
+      # but we can run a little over that limit.
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "c6:96:98:82:dc:04:6c:39:dd:ac:83:05:e3:15:1c:98"
+      - checkout
+      - attach_workspace:
+          at: /go/bin
+      - setup_remote_docker:
+          docker_layer_cache: true
+      - run: &install_gotestsum
+          name: Install gotestsum
+          command: |
+            url=https://github.com/gotestyourself/gotestsum/releases/download
+            curl -sSL "${url}/v${GOTESTSUM_RELEASE}/gotestsum_${GOTESTSUM_RELEASE}_linux_amd64.tar.gz" | \
+              sudo tar -xz --overwrite -C /usr/local/bin gotestsum
+      - run: go mod download
+      - run:
+          name: Waiting for Postgres to be ready
+          command: |
+            for i in `seq 1 10`;
+            do
+              nc -z localhost 5432 && echo Success && exit 0
+              echo -n .
+              sleep 1
+            done
+            echo Failed waiting for Postgres && exit 1
+      - run:
+          name: go test
+          command: |
+            mkdir -p $TEST_RESULTS_DIR
+            PACKAGE_NAMES=$(go list ./... | circleci tests split --split-by=timings --timings-type=classname)
+            echo "Running $(echo $PACKAGE_NAMES | wc -w) packages"
+            echo $PACKAGE_NAMES
+            gotestsum --format=short-verbose \
+              --junitfile $TEST_RESULTS_DIR/gotestsum-report.xml -- \
+              -tags="$GOTAGS" -p 2 \
+              -cover -coverprofile=coverage.txt \
+              $PACKAGE_NAMES
+      - store_test_results:
+          path: *TEST_RESULTS_DIR
+      - store_artifacts:
+          path: *TEST_RESULTS_DIR
+      - notify_main_failure
```

Comment:
This is a step defined by us, not something built in to circle. https://github.com/hashicorp/waypoint/blob/a2bc8b461adace6c720645a02d7c7ca26730912a/.circleci/config.yml#L509-L550
We'll need to make sure this project has a `SLACK_WEBHOOK` defined as well.

Reply:
Good catch! Is there a good way to do that? Is it just something we add in circle once the project workflow has been created? :thinking:

Reply:
Ok, we got this sorted out now I think. We created a new webhook and I just added it to the project on Circle :)